lms/djangoapps/notification_prefs/views.py

```diff
@@ -95,6 +96,8 @@ def enable_notifications(user):
     Enable notifications for a user.
     Currently only used for daily forum digests.
     """
+    # Calling UserPreference directly because this method is called from a couple of places,
+    # and it is not clear that user is always the user initiating the request.
     UserPreference.objects.get_or_create(
```

Comment:
We'd said that we'd always authorize creation of user preferences, so this seems like a bit of an anomaly. I wonder if we can refactor the logic so that the requesting user is available.